lib/src/index_impl.dart

```diff
@@ -79,6 +84,19 @@ class MeiliSearchIndexImpl implements MeiliSearchIndex {
     await http.deleteMethod('/indexes/$uid');
   }
 
+  @override
+  Future<MeiliSearchIndex> fetchInfo() async {
+    final index = await client.getIndex(uid);
+    this.primaryKey = index.primaryKey;
+    return index;
```

Comment:
- use `_primaryKey` instead of `this.primaryKey`
- update `_createdAt` and `_updatedAt`